### ISSUE TITLE
Hexadecimal longs are parsed incorrectly. 

### ISSUE THREAD

**Issue by aaronla-ms** [PowerShell/PowerShell#14302]
## Steps to reproduce

```powershell
echo (0xffffffffl)
```

## Expected behavior

The behavior on Windows Powershell 5.1.19041.610

```none
4294967295
```

## Actual behavior

```none
-1
```

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.3
PSEdition                      Core
GitCommitId                    7.0.3
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by aaronla-ms** [comment]
The issue does appear to be resolved on powershell 7.1.0, but 7.1.0 isn't available on any LTS release of dotnet. Do you know if this was a fix in Powershell or dotnetcore? If the former, it would be great if we could backport this to powershel 7.0.x for use on an LTS release of dotnet.

**Comment by aaronla-ms** [comment]
Issue first seen in 7.0.0. Version 6.2.3 reports the correct behavior.

**Comment by vexx32** [comment]
Hmm, that's interesting, that looks like it's hitting the regular int parsing code rather than the long parsing code as we'd expect. I'll have a quick look.

**Comment by vexx32** [comment]
Oh, right, yeah we fixed this already, I missed that comment. Let me check my PR history. It's not up to me whether it gets backported or not, though, I'm afraid, but we can ask after it.

**Comment by vexx32** [comment]
https://github.com/PowerShell/PowerShell/pull/11844 is the PR that affected and fixed this behaviour.

@SteveL-MSFT is there any chance of backporting this fix to the 7.0.x branch?

**Comment by iSazonov** [comment]
@vexx32 Thanks! I added BackPort-7.0.x-Consider in the PR.
